src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -360,26 +360,23 @@ public void run()
             {
                 try
                 {
-                    fut.handle((ignored, err) -> {
-                        compactingCF.remove(cfs);
-
-                        if (err != null)
-                            handleCompactionError(err, cfs);
-
-                        // Since we have ran at least one task and thus the set of sstables has changed, check
-                        // for new compaction possibilities. We will still do this if they all errored out or
-                        // the cfs is no longer valid, which is not helpful but not a problem.
-                        submitBackground(cfs);
-                        return null;
-                    }).get();
+                    fut.get();
+                    // Since we have ran at least one task and thus the set of sstables has changed, check
+                    // for new compaction possibilities. We will still do this if they all errored out or
+                    // the cfs is no longer valid, which is not helpful but not a problem.
+                    submitBackground(cfs);
```

Comment:
An `allOf` future will throw if any of the individual futures fails. It's okay to schedule or not schedule if all fail, but if at least one succeeds we do need to schedule a new pass.
Let's move this after the catch (otherwise the comment is incorrect too).

Reply:
45a3cfa6d017e38d94b5581d0efe07ddc45bd79d